srcpkgs/helio-sequencer/template

```diff
@@ -0,0 +1,48 @@
+# Template file for 'helio-sequencer'
+pkgname=helio-sequencer
+version=3.12
+revision=1
+_gitrev="61f7db380f083b424418d65f4e26addc1370ddea"
+archs="i686 x86_64"
+hostmakedepends="git pkg-config"
+makedepends="freetype-devel libX11-devel libXinerama-devel libXrandr-devel libXcursor-devel libXcomposite-devel MesaLib-devel libfreeglut-devel libcurl-devel alsa-lib-devel jack-devel libsigc++-devel"
+depends="curl freetype libXinerama libglvnd alsa-lib"
+short_desc="One music sequencer for all major platforms, desktop and mobile"
+maintainer="IFoundSilentHouse <adeptslab@gmail.com>"
+license="GPL-3.0-or-later"
+homepage="https://helio.fm/"
+changelog="https://github.com/helio-fm/helio-sequencer/blob/develop/CHANGELOG.md"
+
+do_fetch() {
+	# commit 61f7db3 is a commit of stable version 3.12
+	# helio-sequencer has thirdparty repos in its' github 
+	# so we have to use this do_fetch() hack
+	git clone --recursive https://github.com/helio-fm/helio-sequencer.git $wrksrc
+	cd $wrksrc
+	git reset --hard ${_gitrev}
+}
+
+do_build() {
+	case "$XBPS_TARGET_MACHINE" in
+		i686) export CONFIG=Release32 ;;
+		x86_64) export CONFIG=Release64 ;;
+	esac
+	cd $wrksrc/Projects/LinuxMakefile/
+	make all strip
+}
+
+do_install() {
+	cd $wrksrc
+	vbin Projects/LinuxMakefile/build/helio
+}
```

Comment:
1. We don't permit checking out source code. Use `distfiles` to fetch a tagged tarball for the package version.
2. You should be using the `gnu-makefile` build style, which will define these functions for you. Control its behavior with `make_build_args`, `make_build_target`, `make_install_args`, `make_install_target`, etc.

Reply:
Is using git --recursive is considered changing source code? It's an official build instructions to fetch it that way. There's no way to get full souce code without --recursive option. Tarball doesn't provide full source

Reply:
And thats the most important thing to know for me - if we can't git --recursive then we won't be able to have this package

Reply:
You can add the submodules to `distfiles`, let it fetch, and move them into place in `post_extract`. Using `git --recursive` is not an option here. Take a look at, *e.g.*, `grpc` to see how submodules are handled.

Reply:
I looked through grpc... i'm sorry i don't think it can be compared with this package. Cause on update it can just do git --recursive and all the checksums of third party repos will be like developer wanted them to be without changing checksums and links manualy for each one. Is it really that undesirable to use custom do_fetch? 

And as for gnu-makefile... I haven't found an option to exclude "install" from the template. Package builds with only ```make```

Reply:
You can override the `do_install` by providing your own, but use the build style so that it properly drives the build.

As for dropping the custom git-based fetching, you can either do things the right way, or we can close the PR.

Reply:
This is also described in https://meator.github.io/xbps-src-tutorials/mini-tutorials/how-to-fetch-git-submodules.html

This website also contains a collections of tutorials for xbps-src that you may be interested in.

Reply:
OMG meator you had what i need in your tutorial. But my eyes was so bad that i couldn't find it and i had to figure it out myself. Sorry guys. i'm new to github and just didn't know that installing tar.gz from any commit is possible!!! I thought it's available only for tags(releases). Now i'm prepearing new commit. I took in consideration everything that ahesford said and this template looks elegant and productive

Reply:
If you haven't already and if you have time to spare, I recommend you reading https://meator.github.io/xbps-src-tutorials/xbps-src-tutorial/index.html. It explains packaging programs with xbps-src. Many problems with your template like cross compiling, build styles, SHLIB dependencies and more are explained in it.